impl/asm/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -63,13 +63,22 @@
     private static final String XML_SUFFIX = ".xml";
     private static final String META_INF_SERVICES_PREFIX = "META-INF/services/";
     private static final String CLASS_FOR_NAME_PRIVATE_METHOD = "org_wildfly_tranformer_asm_classForName_String__boolean_ClassLoader";
+    private static final String CLASS_OBJECT = "java/lang/Class";
+    private static final String FORNAME_METHOD = "forName";
+    private static final String MAP_OBJECT = "java/util/Map";
+    private static final String MAP_PUT_METHOD = "put";
+    private static final String MAP_PUT_METHOD_DESC = "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;";
+    private static final String MAP_GET_METHOD = "get";
+    
     private static final boolean useASM7 = getMajorJavaVersion() >= 11;
     private boolean classTransformed;
     private boolean alreadyTransformed;
     private String changeClassName;
     final Map<String, String> mappingWithSeps;
     final Map<String, String> mappingWithDots;
     final Set<String> generatedReflectionModelHandlingCode = new HashSet<>();
+    private byte[] generatedReflectionModelHandlingByteCode;
+    private String generatedReflectionModelHandlingClassName;
```

Comment:
Same as above variable - this is not thread safe - shouldn't it be ThreadLocal ?

Reply:
Moved generatedReflectionModelHandlingClassName to be local parameter.

Reply:
@ropalka Note that we still have other transformer state as well which we probably could also make local parameters or something like that:
    private boolean classTransformed;
    private String changeClassName;
These weren't added as part of handling reflection, so probably can be part of a separate later change (I'd do it now, but I also need to see why the ASM transformer isn't being used or isn't generated extra classes from the galleon-plugins (I did update the galleon-plugins to use <artifactId>asm-based-transformer</artifactId>).  Will try again.